File: web/src/app/experience/experience.ts

import {MonthYear} from "../shared/models/month-year";
import {Image} from "../shared/models/image";
import {Location} from "../shared/models/location";

export interface Experience {
  id: number;
  company: string;
  description: string;
  end: MonthYear;
  image: Image;
  location: Location;
  project: string;
  role: string;
  start: MonthYear;
  type: EmploymentType;
}

export enum EmploymentType {
  Contract = "Contract",
  FullTime = "Full-time",
  Seasonal = "Seasonal",
}

File: app/src/main/java/com/wheeler/dao/model/Experience.java
package com.wheeler.dao.model;

import com.wheeler.dao.constant.ExperienceType;
import lombok.Data;

@Data
public class Experience {

    private String id;
    private String name;
    private String description;
    private MonthYear end;
    private Location location;
    private String project;
    private String role;
    private MonthYear start;
    private String title;
    private ExperienceType type;
}